classes/Rest/Controllers/WarehouseControllerProvider.php

```diff
@@ -1401,18 +1401,18 @@ private function getJobData(Application $app, XDUser $user, $realm, $jobId, $act
      */
     private function getJobDataSet(XDUser $user, $realm, $jobId, $action)
     {
-        $configFile = new XdmodConfiguration(
-            'rawstatistics.json',
-            CONFIG_DIR,
-            null,
-            array(
-                'local_config_dir' => implode(DIRECTORY_SEPARATOR, array(CONFIG_DIR, 'rawstatistics.d'))
+        $rawstats = XdmodConfiguration::assocArrayFactory('rawstatistics.json', CONFIG_DIR);
+
+        $realmExists = count(
```

Comment:
Disadvantage of this code is that you always build an array with all of the realms in it. Even though the code can stop looking when a match is found. Also there is no need to create a temporary array.

```suggestion
        $realmExists = false;
        foreach ($rawstats as $item) {
            if ($item['name'] === $realm) {
                $realmExists = true;
                break;
            }
        }
```

```